### PR TITLE
Add new annual baseload breakdown service

### PR DESCRIPTION
**Pull request by iant**

### CODE REVIEW

**Comment by iant** [comment]
hi @ldodds i've removed pricing and co2 so this is ready for review when you get a mo

**Comment by iant** [comment]
@ldodds could you review but not merge please.  I want to double check something with the tests first.

**Comment by ldodds** [comment]
@iant looks OK, but you need to update the tests to include the expected values for each year.

As discussed last week, run the existing analytics test framework for the AlertElectricityBaseloadVersusBenchmark with a couple of different asof_dates, e.g. end of 2022, end of 2021 to generate the YAML that contains the current values produced for the baseload for those years. 

You can then use these as the expected values in the specs.

File: spec/app/services/baseload/baseload_annual_breakdown_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  context '#annual_baseload_breakdowns' do
+    it 'runs the calculation' do
+      annual_baseload_breakdowns = service.annual_baseload_breakdowns
+      expect(annual_baseload_breakdowns.size).to eq(4)
+      expect(annual_baseload_breakdowns.map(&:class).uniq).to eq([Baseload::AnnualBaseloadBreakdown])
+      expect(annual_baseload_breakdowns[0].as_json).to eq(
+        {
+          'average_annual_baseload_kw' => -0.0,
```
Comment (iant) [inline spec/app/services/baseload/baseload_annual_breakdown_service_spec.rb:20]: `ANALYTICSTESTDIR=test_output bundle exec ruby script/standard/test_alerts.rb` with `asof_date = Date.new(2018, 12, 31)`
relevant line in `Acme Academy 20181231 AlertElectricityBaseloadVersusBenchmark raw_variables_for_saving.yaml`:
```
AlertElectricityBaseloadVersusBenchmark:average_baseload_last_year_kw: -0.0
```

File: spec/app/services/baseload/baseload_annual_breakdown_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          'meter_data_available_for_full_year' => false,
+          'year' => 2019
+        }
+      )
+      expect(annual_baseload_breakdowns[1].as_json).to eq(
+        {
+          'average_annual_baseload_kw' => 27.220618980169984,
```
Comment (iant) [inline spec/app/services/baseload/baseload_annual_breakdown_service_spec.rb:27]: `ANALYTICSTESTDIR=test_output bundle exec ruby script/standard/test_alerts.rb` with `asof_date = Date.new(2019, 12, 31)`
relevant line in `Acme Academy 20191231 AlertElectricityBaseloadVersusBenchmark raw_variables_for_saving.yaml`:
```
AlertElectricityBaseloadVersusBenchmark:average_baseload_last_year_kw: 27.220618980169984
```

File: spec/app/services/baseload/baseload_annual_breakdown_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      expect(annual_baseload_breakdowns[2].as_json).to eq(
+        {
+          'average_annual_baseload_kw' => 26.174109589041098,
+          'meter_data_available_for_full_year' => true,
+          'year' => 2021
+        }
+      )
+      expect(annual_baseload_breakdowns[3].as_json).to eq(
+        {
+          'average_annual_baseload_kw' => 24.5531506849315,
```
Comment (iant) [inline spec/app/services/baseload/baseload_annual_breakdown_service_spec.rb:41]: `ANALYTICSTESTDIR=test_output bundle exec ruby script/standard/test_alerts.rb` with `asof_date = Date.new(2021, 12, 31)`
relevant line in `Acme Academy 20211231 AlertElectricityBaseloadVersusBenchmark raw_variables_for_saving.yaml`:
```
AlertElectricityBaseloadVersusBenchmark:average_baseload_last_year_kw: 24.5531506849315
```

**Pull request merged**